test/sources/npm_test.rb

```diff
@@ -80,5 +80,16 @@
         end
       end
     end
+
+    describe "missing dependencies" do
```

Comment:
This should also include a test that the non-yarn error scenario still occurs when `yarn.lock` is not present 🙇

Reply:
Added & generally improved those tests: they make sure that `yarn.lock` is present or missing (depending on method).